### ISSUE TITLE
documentation for running rake resource:generate

### ISSUE THREAD

**Issue by itewk** [rhtconsulting/puppet-jboss_admin#34]
When I try to run rake resource:generate I get an error:

```
[ian@IanT-RedHat tasks]$ rake resource:generate --trace
(in /home/ian/Projects/USPTO/puppet/PSB-JBOSS-ADMIN-MODULE)
rake aborted!
LoadError: cannot load such file -- lib/tasks/resource.rb
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
/home/ian/Projects/USPTO/puppet/PSB-JBOSS-ADMIN-MODULE/Rakefile:4:in `<top (required)>'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/rake_module.rb:28:in `load'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/rake_module.rb:28:in `load_rakefile'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/application.rb:677:in `raw_load_rakefile'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/application.rb:94:in `block in load_rakefile'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/application.rb:176:in `standard_exception_handling'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/application.rb:93:in `load_rakefile'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/application.rb:77:in `block in run'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/application.rb:176:in `standard_exception_handling'
/home/ian/.gem/ruby/gems/rake-10.2.2/lib/rake/application.rb:75:in `run'
/usr/bin/rake:37:in `<main>'
```

I assume this is probably because I have not installed all the required gems I need installed. I attempted installing all the gems marked as required in the various lib/tasks files but not all of them I can install.

I was able to install
* json
* puppet-lint
* puppetlabs_spec_helper

I was not able to figure out how to install
* rubygems
* erb
* backports/1.9.1/symbol/comparison

**Comment by simhpl** [comment]
Hi!

How to solve this?
I have very similar error:

```
rake generate:resources --trace
rake aborted!
LoadError: cannot load such file -- lib/tasks/resource.rb
puppet-jboss_admin/Rakefile:4:in `<top (required)>'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/rake_module.rb:28:in `load'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/rake_module.rb:28:in `load_rakefile'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:677:in `raw_load_rakefile'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:94:in `block in load_rakefile'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:176:in `standard_exception_handling'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:93:in `load_rakefile'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:77:in `block in run'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:176:in `standard_exception_handling'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:75:in `run'
/var/lib/gems/2.1.0/gems/rake-10.2.2/bin/rake:33:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'
```

I get the same on Fedora 22 and Kubuntu 15.04.

**Comment by itewk** [comment]
@simhpl 

It maybe as simple as the following:

> gem install top

**Comment by simhpl** [comment]
@itewk 
If it's required shouldn't bundler install it?

Ayways, it didn't help.

```
$ rake generate:resources --trace
rake aborted!
LoadError: cannot load such file -- lib/tasks/resource.rb
/usr/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/usr/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
puppet-jboss_admin/Rakefile:4:in `<top (required)>'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/rake_module.rb:28:in `load'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/rake_module.rb:28:in `load_rakefile'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:677:in `raw_load_rakefile'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:94:in `block in load_rakefile'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:176:in `standard_exception_handling'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:93:in `load_rakefile'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:77:in `block in run'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:176:in `standard_exception_handling'
/var/lib/gems/2.1.0/gems/rake-10.2.2/lib/rake/application.rb:75:in `run'
/var/lib/gems/2.1.0/gems/rake-10.2.2/bin/rake:33:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'
```

**Comment by itewk** [comment]
@simhpl 
I am actually not currently set up to try this out myself but talking to one of my fellow cohorts we think the problem maybe with the require statement in Rakefile.

Try updating the following:
```
require 'lib/tasks/resource.rb'
```
to
```
require_relative 'lib/tasks/resource.rb'
```

What version of Ruby are you running?

**Comment by itewk** [comment]
@cpitman can you take this?

**Comment by itewk** [comment]
I resolved the generation in issue #104

**Comment by itewk** [comment]
added documentation for setting up your environment for this to the Readme.
